src/fish_indent.cpp

```diff
@@ -251,9 +252,28 @@ void prettifier_t::prettify_node(const parse_node_tree_t &tree, node_offset_t no
                 if (prev_node_type != parse_token_type_redirection) {
                     append_whitespace(node_indent);
                 }
-                output.append(source, node.source_start, node.source_length);
+                wcstring unescaped{source, node.source_start, node.source_length};
+                // Unescape the string - this leaves special markers around if there are any expansions or anything.
+                // TODO: This also already computes backslash-escapes like \u or \x.
+                // We probably don't want that - if someone picked `\x20` to spell space, they have a reason.
+                unescape_string_in_place(&unescaped, UNESCAPE_SPECIAL);
+
+                // Remove INTERNAL_SEPARATOR because that's a quote.
+                auto quote = [](const wchar_t ch) {
+                                 return ch == INTERNAL_SEPARATOR;
+                             };
+                unescaped.erase(std::remove_if(unescaped.begin(), unescaped.end(), quote), unescaped.end());
+
+                // If no non-alphanumeric char is left, use the unescaped version.
+                // This can be extended to other characters, but giving the precise list is tough,
+                // can change over time (see "^", "%" and "?", in some cases "{}") and it just makes people feel more at ease.
+                if (std::find_if_not(unescaped.begin(), unescaped.end(), fish_iswalnum) == unescaped.end() && !unescaped.empty()) {
```

Comment:
Can we add just `_` and `-` to the allowed characters? Since they are commonly used for compound names.

Reply:
I'd like it if we could anchor this to some other character set. I don't think we have a set of "safe" characters anywhere. We could also pick valid variable names, which is this plus "_", but not "-".

Note that this would have real impact on how our completions format - there are some that have "kebab-case" commands.

Or we just declare a random set of valid characters, that wouldn't be too awful.

Reply:
Okay, I pushed an additional list of "-", "_" and "/". To show the difference I added another commit just with those changes, but I'll remove that when merging.